docker-entrypoint.sh

```diff
@@ -35,7 +35,7 @@ generate_prefs_init() {
 
   local email_verification='true'
   local auth_providers='["localdb"]'
-  local auth_configs=(/data/xnat/home/config/auth/!(ldap-provider.properties.example))
+  local auth_configs=($(find /data/xnat/home/config/auth/ -type f -name "*.properties"))
```

Comment:
You can use a bash glob here instead of creating a new process! Silly little optimization, but fun to know!

https://mywiki.wooledge.org/ParsingLs

```
setopt -s nullglob # at the top of the file with the other options
local auth_configs=( /data/xnat/home/config/auth/*.properties )
```

This is great as is though!

Reply:
Ooh, neat! Now I'm beginning to see why this may have broken in my case.  Please correct me if I'm wrong.  

At the top of the script, we have `#shopt -s extglob`.  In the docs you sent, the author advises us to: `Consider also using "shopt -s nullglob" so that an empty directory won't give you a literal '*'.`

My theory is that since I was not mounting the volume pointing to the template config, the bash glob was returning the literal pattern string.  I thought it was enough to COPY in the Dockerfile, but from my testing, it's evident that those files still need to be mounted inside of the container via the docker-compose.yaml volumes. 

```
    volumes:
       - "${PWD}/ldap-provider.properties.example:/data/xnat/home/config/auth/ldap-provider.properties.example"
```

Essentially, the quickest fix would have been to use `nullglob` instead of `extglob`...  Does this seem correct?